packages/dd-trace/test/appsec/iast/analyzers/set-cookies-header-interceptor.spec.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const setCookiesHeaderInterceptor = require('../../../../src/appsec/iast/analyzers/set-cookies-header-interceptor')
+const dc = require('../../../../../diagnostics_channel')
+
+const iastSetCookieChannel = dc.channel('datadog:iast:set-cookie')
+const setHeaderChannel = dc.channel('datadog:http:server:response:set-header:finish')
+
+describe('Test IntermediateCookiesAnalyzer', () => {
+  let setCookieCallback
+
+  beforeEach(() => {
+    setCookieCallback = sinon.stub()
+    setCookiesHeaderInterceptor.configure(true)
+    iastSetCookieChannel.subscribe(setCookieCallback)
+  })
+
+  afterEach(() => {
+    setCookiesHeaderInterceptor.configure(false)
+    iastSetCookieChannel.unsubscribe(setCookieCallback)
+  })
+
+  it('Should send set-cookie-event if the header is "set-cookie"', () => {
+    setHeaderChannel.publish({
+      name: 'set-cookie',
+      value: 'key=value; Secure; HttpOnly',
```

Comment:
What will happen with a value like: `key=value1;value2;orValueWith=; Secure; HttpOnly`?

Reply:
It will set only one cookie, as secure and we are not going to report it. This si correct.

To write multiple cookies, you should do something like:
```
res.setHeader('set-cookie',['key=value1', 'otherKey=secure; Secure'])
```

Reply:
I mean a cookie value containing `;` or `=` chars

```
res.setHeader('set-cookie', 'key=value_with_;_or_=_for_annoyance')
```

Reply:
It is not supported by cookie
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

> A <cookie-value> can optionally be wrapped in double quotes and include any US-ASCII character excluding control characters (ASCII characters 0 up to 31 and ASCII character 127), [Whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace), double quotes, commas, **semicolons**, and backslashes.